libraries/redcore/model/list.php

```diff
@@ -228,129 +228,140 @@ public function getPagination()
 	 */
 	protected function populateState($ordering = null, $direction = null)
 	{
-		// If the context is set, assume that stateful lists are used.
-		if ($this->context)
+		// If no context is set, assume that stateful lists are not used.
+		if (!$this->context)
 		{
-			$app = JFactory::getApplication();
+			$this->setState('list.start', 0);
+			$this->setState('list.limit', 0);
 
-			// Receive & set filters
-			if ($filters = $app->getUserStateFromRequest($this->context . '.filter', 'filter', array(), 'array'))
-			{
-				foreach ($filters as $name => $value)
-				{
-					$this->setState('filter.' . $name, $value);
-				}
-			}
+			return;
+		}
 
-			$limit = 0;
+		$app = JFactory::getApplication();
 
-			// Receive & set list options
-			if ($list = $app->getUserStateFromRequest($this->context . '.list', 'list', array(), 'array'))
-			{
-				foreach ($list as $name => $value)
-				{
-					// Extra validations
-					switch ($name)
-					{
-						case 'fullordering':
-							$orderingParts = explode(' ', $value);
+		$data = array(
+			'filter' => $app->getUserStateFromRequest($this->context . '.filter', 'filter', array(), 'array'),
+			'list'   => $app->getUserStateFromRequest($this->context . '.list', 'list', array(), 'array')
+		);
 
-							if (count($orderingParts) >= 2)
-							{
-								// Latest part will be considered the direction
-								$fullDirection = end($orderingParts);
+		$filterForm = $this->getForm($data, false);
 
-								if (in_array(strtoupper($fullDirection), array('ASC', 'DESC', '')))
-								{
-									$this->setState('list.direction', $fullDirection);
-								}
+		$data = $filterForm ? $this->validate($filterForm, $data) : array();
 
-								unset($orderingParts[count($orderingParts) - 1]);
+		$app->setUserState($this->context, $data);
 
-								// The rest will be the ordering
-								$fullOrdering = implode(' ', $orderingParts);
+		$limit = 0;
 
-								if (in_array($fullOrdering, $this->filter_fields))
-								{
-									$this->setState('list.ordering', $fullOrdering);
-								}
-							}
-							else
-							{
-								$this->setState('list.ordering', $ordering);
-								$this->setState('list.direction', $direction);
-							}
-							break;
+		// Data has been filtered and validated by the form. Populate state
+		if ($data)
+		{
+			$filters = empty($data['filter']) ? array() : $data['filter'];
```

Comment:
This is still needed. The array is indeed started from request and initialised to array but then is passed and filtered by the form. That means that if the form has no filters section `$data['filters']` won't be set,